rclcpp/include/rclcpp/subscription_traits.hpp

```diff
@@ -0,0 +1,79 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__SUBSCRIPTION_TRAITS_HPP_
+#define RCLCPP__SUBSCRIPTION_TRAITS_HPP_
+
+#include <memory>
+
+#include "rclcpp/function_traits.hpp"
+
+namespace rclcpp
+{
+namespace subscription_traits
+{
+
+/*
+ * The current version of uncrustify has a misinterpretion here
+ * between `:` used for inheritance vs for initializer list
+ * The result is that whenever a templated struct is used,
+ * the colon has to be without any whitespace next to it whereas
+ * when no template is used, the colon has to be separated by a space.
+ * Cheers!
+ */
+template<typename T>
+struct is_raw_subscription_argument : std::false_type
```

Comment:
Maybe if you used the `public` keyword before the inheritance uncrustify would do the right thing?

```c++
template<typename T>
struct is_raw_subscription_argument: public std::false_type
{};
```

Reply:
Unfortunately, it still complains:
```
31: +++ include/rclcpp/subscription_traits.hpp.uncrustify
31: @@ -40 +40 @@
31: -struct is_raw_subscription_argument<rcl_message_raw_t> : public std::true_type
31: +struct is_raw_subscription_argument<rcl_message_raw_t>: public std::true_type
```

Good idea though!